### PR TITLE
Implement fake tags and addTag/deleteTag

### PR DESCRIPTION
**Pull request by puiterwijk**
Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>

### CODE REVIEW

**Comment by bowlofeggs** [comment]
The test failure was unrelated to this patch, but required the patch to be rebased onto the latest develop with the fix. I've force pushed your branch to fix that.

**Review by bowlofeggs — APPROVED** [review]

File: bodhi/server/buildsys.py [hunk 1]
```diff
@@ -428,6 +443,26 @@ def getTag(self, taginfo, **kw):
                 'perm': None, 'id': 246, 'arches': None,
                 'maven_include_all': False, 'perm_id': None}
 
+    def addTag(self, tag, **opts):
+        """ Emulate tag adding. """
```
Comment (bowlofeggs) [inline bodhi/server/buildsys.py]: The test failure was due to the spaces around this sentence:

```
rawhide	bodhi/server/buildsys.py:446 in public method `addTag`:
rawhide	        D210: No whitespaces allowed surrounding docstring text
```

File: bodhi/server/buildsys.py [hunk 2]
```diff
@@ -428,6 +443,26 @@ def getTag(self, taginfo, **kw):
...
+        opts['maven_support'] = False
+        opts['name'] = tag
+        opts['perm'] = 'admin'
+        opts['arches'] = None
+        opts['maven_include_all'] = False
+        opts['perm_id'] = 1
+        self.__tags__.append((tag, opts))
+
+    def deleteTag(self, tagid):
+        """ Emulate tag deletion. """
```
Comment (bowlofeggs) [inline bodhi/server/buildsys.py]:
```
rawhide	bodhi/server/buildsys.py:462 in public method `deleteTag`:
rawhide	        D210: No whitespaces allowed surrounding docstring text
```

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
Just two docblock changes needed to satisfy pydocstyle.

File: bodhi/server/buildsys.py [hunk 3]
```diff
@@ -119,6 +119,14 @@ def getTag(self, *args, **kw):
         """Raise NotImplementedError."""
         raise NotImplementedError
 
+    def addTag(self, *args, **kw):
+        """Raise NotImplementedError."""
+        raise NotImplementedError
+
+    def deleteTag(self, *args, **kw):
+        """Raise NotImplementedError."""
+        raise NotImplementedError
```
Comment (bowlofeggs) [inline bodhi/server/buildsys.py:128]: Hmm, looks like CI is upset about missing coverage on the above two methods. I added a really [dumb test](https://github.com/fedora-infra/bodhi/blob/3.6.1/bodhi/tests/server/test_buildsys.py#L32-L43) to hit the other methods here, so you could just add to that.

I also think it's reallllly dumb that we have this class at all. We could also just kill the entire class if you want. I really don't see what purpose it serves - it's not like Koji subclasses it - only ```DevBuildsys``` does, and it has its own methods anyway.

Either of the above solutions are fine by me.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]

File: bodhi/server/buildsys.py (hunk 3 above)
Comment (puiterwijk) [reply bodhi/server/buildsys.py:128]: Just to note: I've went for the dumb test method.

**Review by bowlofeggs — APPROVED** [review]

**Comment by bowlofeggs** [comment]
This patch is planned to be included in the upcoming 3.9.0 release: #2477.

**Pull request merged**